src/providers/BasicPoolsProvider.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  ChainId,
+  POOLS_MAP,
+  PoolName,
+  PoolTypes,
+  Token,
+  getIsLegacySwapABIPoolByAddress,
+  getMinichefPid,
+} from "../constants"
+import React, { ReactElement, useEffect, useState } from "react"
+import { getContract, getSwapContract } from "../utils"
+
+import { AppState } from "../state"
+import { BigNumber } from "@ethersproject/bignumber"
+import ERC20_ABI from "../constants/abis/erc20.json"
+import { Erc20 } from "./../../types/ethers-contracts/Erc20.d"
+import { MetaSwap } from "../../types/ethers-contracts/MetaSwap"
+import { Web3Provider } from "@ethersproject/providers"
+import { getMigrationData } from "../utils/migrations"
+import { useActiveWeb3React } from "../hooks"
+import { useSelector } from "react-redux"
+
+type SwapInfo = {
+  poolAddress: string
+  lpToken: string
+  typeOfAsset: PoolTypes
+  poolName: string
+  targetAddress: string | null
+  tokens: string[]
+  underlyingTokens: string[] | null
+  basePoolAddress: string | null
+  metaSwapDepositAddress: string | null
+  isSaddleApproved: boolean
+  isRemoved: boolean
+  isGuarded: boolean
+  isPaused: boolean
+  virtualPrice: BigNumber
+  adminFee: BigNumber
+  swapFee: BigNumber
+  aParameter: BigNumber
+  tokenBalances: BigNumber[]
+  underlyingTokenBalances: BigNumber[] | null
+  lpTokenSupply: BigNumber
+  miniChefRewardsPid: number | null
+  isMetaSwap: boolean
+}
+
+type BasicPool = {
+  isMigrated: boolean
+  sdlPerDay: BigNumber
+} & SwapInfo
+type BasicPools = Record<string, BasicPool> | null
+
+export const BasicPoolsContext = React.createContext<BasicPools>(null)
+
+export default function BasicPoolsProvider({
+  children,
+}: React.PropsWithChildren<unknown>): ReactElement {
+  const { chainId, library } = useActiveWeb3React()
+  const [basicPools, setBasicPools] = useState<BasicPools>(null)
```

Comment:
Is there any advantage from the UX side to providing defaults here? I'm not sure if the user will experience any improvement if there is defaults or not.

Reply:
This will all be replaced with registry calls, so basically if the registry calls fail, then the user can't see any pools. Thus I'm fine having the default be null instead of some half-filled pool representations